### DISCUSSION TITLE
I'm wondering if there is a configuration for stoping the script when a failure happens?

### DISCUSSION THREAD

**Discussion by neil-armstrong-instil** [Playful-Systems/scriptful#3] [General]
I have a generic "build" script which runs the build in each of my projects in a monorepo and I want to it fail if any of the builds fail. I have it setup like this:
```
  build: sequential([
    command({
      run: "yarn build",
      cwd: "shared"
    }),
    parallel([
      command({
        run: "yarn build",
        cwd: "webapp"
      }),
      command({
        run: "yarn build",
        cwd: "electron"
      }),
      command({
        run: "yarn build",
        cwd: "acceptance-tests"
      }),
    ])
  ]),
```

So it will run the "shared" folder first, followed by webapp, electron, and acceptance tests in parallel. However when I force the "shared" folder to fail (via a basic test that just throws an error) it will just carry on with the rest of the build commands and not "fail the build". I also presume github actions won't be aware of the failure either which means I'd sadly still need to provide a package.json flavor of this command for CI purposes :(.

I can have a poke around to create a pull request if this is something that doesn't exist yet as I do quite like the idea of the library!

**Comment by NWylynko** [comment]
Sorry I didn't see this earlier, I have already made an issue for this and will get started with it when I get home.

**Comment by NWylynko** [comment]
Hey @neil-armstrong-instil if you've got a minute I would love your opinion. Do you feel the default should be to continue or stop on fail? I think it should be to stop, and configurable to continue if people want that.

**Reply by neil-armstrong-instil** [reply]
I'd go with fail as it's what you think is best, and you are the creator so will always be the primary user :). [Dogfood](https://en.wikipedia.org/wiki/Eating_your_own_dog_food) it and see if it works!

**Comment by NWylynko** [comment]
Sorry couldn't get this done tonight, weird issues with catching the error. I will pick this back up tomorrow.

**Reply by neil-armstrong-instil** [reply]
No rush at all! Take your time, or if you don't have time I can have a look around myself 👍

**Comment by NWylynko** [comment]
So the underlying issue was on exit / close of the child process, it wasn't checking the status code to see if it was a success or not. In version 1.2.0 it now checks the status code and rejects appropriately. This fixes your issue as the process will stop if the shared module fails to build. But for some reason the trycatch fails to catch the promise being thrown so controlling what should happen on failure is difficult.

**Comment by NWylynko** [comment]
I have just pushed version 1.2.1, adding in an optional parameter to the command() function called 'ContinueOnFailure' setting this to true well then essentially revert back to the pre 1.2.0 implementation of continuing on to the next steps even if the step fails. This defaults to false, ending the flow in the case that a step fails.
